SwissDataScienceCenter/renku-ui
# Allow remembering storage credentials as secrets

# Changes

This PR supports
- Saving credentials for cloud storage as secrets when starting a session
- Viewing credential secrets on the secrets page.

# Testing

After creating a project that includes Cloud Storage which requires credentials (e.g., a private S3 bucket) and a Session, when you start the session on this project, you will be prompted to provide your credentials with an option to save them.

<img width="472" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/8b45dd11-9881-4ed7-9157-ac36d3d5546b">

Once the session has been launched, it is possible to go to the Secrets (V1) page and see the secrets for the storage.

<img width="1236" alt="image" src="https://github.com/SwissDataScienceCenter/renku-ui/assets/1196411/7bced4c9-0d5d-4cfc-a2ec-541081dc27e9">

Some scenarios to test include:

- start a session skipping credentials and make sure it starts
- start a session, saving credentials
- check that the saved credentials are visible in the Secrets page
- check that after saving credentials, they are no longer necessary for starting a session (the session will not actually start because the backend is not yet available)
- check that one of the storage credentials can be deleted and then you should not be prompted. 

/deploy renku-data-services=pitch/save-storage-credentials-api
